src/io.jl

```diff
@@ -227,3 +297,116 @@ hg_load(
     V = Nothing,
     E = Nothing) where {U <: Real} =
     open(io -> hg_load(io, format; T=T, D=D, V=V, E=E), fname, "r")
+
+
+
+"""
+    hg_load(
+        io::IO,
+        format::HIF_Format;
+        T::Type{U} = Bool,
+        D::Type{<:AbstractDict{Int, U}} = Dict{Int,U},
+        V::Union{Type{String}, Type{Int}} = String,
+        E::Union{Type{String}, Type{Int}} = String
+    ) where {U <: Real}
+
+Loads a hypergraph from a stream `io` from `HIF` format.
+More info: https://github.com/pszufe/HIF-standard
+
+**Arguments**
+
+* `T` : type of weight values stored in the hypergraph's adjacency matrix
+* `D` : dictionary for storing values the default is `Dict{Int, T}`
+* `V` : type of values stored in the vertices of the hypergraph
+* `E` : type of values stored in the edges of the hypergraph
+
+"""
+function hg_load(
+    io::IO,
+    format::HIF_Format;
+    T::Type{U} = Bool,
+    D::Type{<:AbstractDict{Int, U}} = Dict{Int, T},
+    V::Union{Type{String}, Type{Int}} = String,
```

Comment:
since `V` and `E` are parametric the code should perform type casting when collecting their values from JSON.
So this should be `V(val)` or `E(val)`. For string you will need to handle this separately as for the `String` target the `string` method should be called.